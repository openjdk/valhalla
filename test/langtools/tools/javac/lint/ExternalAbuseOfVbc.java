/*
 * @test /nodynamiccopyright/
 * @bug 8254274
 * @summary lint should warn when an instance of a value based class is synchronized upon
 * @compile/fail/ref=ExternalAbuseOfVbc.out -XDrawDiagnostics -Werror -Xlint:synchronize ExternalAbuseOfVbc.java
 * @compile/ref=LintModeOffAbuseOfVbc.out -XDrawDiagnostics -Werror -Xlint:-synchronize ExternalAbuseOfVbc.java
 */

public final class ExternalAbuseOfVbc {

    final Integer val = Integer.valueOf(42);
    final String ref = "String";

    void abuseVbc() throws InterruptedException {

        synchronized(ref) {      // OK
            synchronized (val) { // WARN

                ref.wait();      // OK
                val.wait();      // WARN

                ref.notify();    // OK
                val.notify();    // WARN

                ref.notifyAll(); // OK
                val.notifyAll(); // WARN

                ref.wait(10);    // OK
                val.wait(10);    // WARN

                ref.wait(Integer.valueOf(10)); // OK
                val.wait(Integer.valueOf(10)); // WARN

                ref.wait(Long.valueOf(10));    // OK
                val.wait(Long.valueOf(10));    // WARN

                ref.wait(10L);  // OK
                val.wait(10L);  // WARN

                ref.wait(10L, 10); // OK
                val.wait(10L, 10); // WARN

                ref.wait(Long.valueOf(10), Integer.valueOf(10)); // OK
                val.wait(Long.valueOf(10), Integer.valueOf(10)); // WARN
            }
        }
    }
}

