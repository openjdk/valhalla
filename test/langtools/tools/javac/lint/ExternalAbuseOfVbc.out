ExternalAbuseOfVbc.java:17:13: compiler.warn.attempt.to.synchronize.on.instance.of.value.based.class
ExternalAbuseOfVbc.java:20:25: compiler.warn.attempt.to.synchronize.on.instance.of.value.based.class
ExternalAbuseOfVbc.java:23:27: compiler.warn.attempt.to.synchronize.on.instance.of.value.based.class
ExternalAbuseOfVbc.java:26:30: compiler.warn.attempt.to.synchronize.on.instance.of.value.based.class
ExternalAbuseOfVbc.java:29:25: compiler.warn.attempt.to.synchronize.on.instance.of.value.based.class
ExternalAbuseOfVbc.java:32:25: compiler.warn.attempt.to.synchronize.on.instance.of.value.based.class
ExternalAbuseOfVbc.java:35:25: compiler.warn.attempt.to.synchronize.on.instance.of.value.based.class
ExternalAbuseOfVbc.java:38:25: compiler.warn.attempt.to.synchronize.on.instance.of.value.based.class
ExternalAbuseOfVbc.java:41:25: compiler.warn.attempt.to.synchronize.on.instance.of.value.based.class
ExternalAbuseOfVbc.java:44:25: compiler.warn.attempt.to.synchronize.on.instance.of.value.based.class
- compiler.err.warnings.and.werror
1 error
10 warnings
